common/hyperscaler/rules/rules_service.go

```diff
@@ -23,5 +32,104 @@ func NewRulesServiceFromFile(rulesFilePath string) (*RulesService, error) {
 		return nil, err
 	}
 
-	return &RulesService{}, nil
+	rs := &RulesService{
+		parser:    &SimpleParser{},
+		sort:      sort,
+		unique:    unique,
+		signature: signature,
+	}
+
+	rs.Parsed = rs.parse(rulesConfig)
+	return rs, err
+}
+
+func NewRulesServiceFromString(rules string, sort, unique, signature bool) (*RulesService, error) {
+	entries := strings.Split(rules, ";")
+
+	rulesConfig := &RulesConfig{
+		Rules: entries,
+	}
+
+	rs := &RulesService{
+		parser:    &SimpleParser{},
+		sort:      sort,
+		unique:    unique,
+		signature: signature,
+	}
+
+	rs.Parsed = rs.parse(rulesConfig)
+	return rs, nil
+}
+
+func (rs *RulesService) parse(rulesConfig *RulesConfig) *ParsingResults {
+	results := NewParsingResults()
+
+	for _, entry := range rulesConfig.Rules {
+		rule, err := rs.parser.Parse(entry)
+
+		results.Apply(entry, rule, err)
+	}
+
+	if rs.sort {
+		results.Results = SortRuleEntries(results.Results)
+	}
+
+	if rs.unique {
+		results.CheckUniqueness()
+	}
+
+	if rs.signature {
+		results.CheckSignatures()
+	}
+
+	if rs.sort {
+		results.Results = SortRuleEntries(results.Results)
+	}
+
+	return results
+}
+
+func (rs *RulesService) Match(data *MatchableAttributes) map[uuid.UUID]*MatchingResult {
```

Comment:
Very cryptic.
What we match with what?
Why we return map with UUID as keys?

Reply:
In CLI we need keep matching together but in KEB matching will be separated. I used uuid as simple unique key generator to reference rules between processing and matching results.

Reply:
So currently MatchableAttributes consists of plan, platform region and hyperscaler region.
These are all pieces of data being used to determine proper secret binding pool.
Still not seeing any UnMatchableAttributes.
Hence I would propose: ProvisioningAttributes.